### PR TITLE
Lift queue length limit and add warning about long queue complaints

### PR DESCRIPTION
**Pull request by Broxzier**
This PR removes the hardcoded limit of 1000 quests per queue, and instead leaves it up to the player how to deal with long queues entirely. To keep novice players aware of the bad design of extremely long queues, a new warning message is added. The attached park will trigger this warning in just a few seconds.

[long-queue-test.zip](https://github.com/OpenRCT2/OpenRCT2/files/8412143/long-queue-test.zip)

The message is of Ride type, because if it was set to guests' "I've been queueing for ride X for ages" thought it would display none, since that thought is linked to a specific ride (which cannot be linked to a news item currently). Instead, the linked ride is the one with the most complaints. Pretty sure this is usually just one anyway.

Since this PR removes the new cheat, its entry is removed from the changelog.

This game design suggestion was made while https://github.com/OpenRCT2/OpenRCT2/pull/16855 was open. Thanks to that PR, long queues will not cause an overflow issue in the ride window.

### CODE REVIEW

**Comment by karst** [comment]
Perhaps it's still worthy to mention in the changelog: "[Change] Queue limit is now infinite, rather than 1000" or something alike?

**Comment by Broxzier** [comment]
@karst Looks like you were just a bit too fast 😄 I created this PR to get a valid issue/pull ID for it first.

File: src/openrct2/entity/Peep.cpp [hunk 1]
```diff
@@ -1146,15 +1166,33 @@ void peep_problem_warnings_update()
...
         if (gConfigNotifications.guest_warnings)
         {
-            News::AddItemToQueue(News::ItemType::Peeps, STR_PEEPS_GETTING_LOST_OR_STUCK, 16, {});
+            constexpr auto thoughtId = static_cast<uint32_t>(PeepThoughtType::Lost);
+            News::AddItemToQueue(News::ItemType::Peeps, STR_PEEPS_GETTING_LOST_OR_STUCK, thoughtId, {});
+        }
+    }
+
+    if (warning_throttle[7])
+        --warning_throttle[7];
```
Comment (Broxzier) [inline src/openrct2/entity/Peep.cpp:1184]: Just to clarify on this hardcoded index, index 7 seems to be unused before this PR. The array `gPeepWarningThrottle` has a length of 16, but only the first 7 are being used. This array is not accessed anywhere else, other than for saving and loading.

File: distribution/changelog.txt [hunk 2]
```diff
@@ -36,6 +35,7 @@
 - Improved: [#16408] Improve --version cli option to report more compatibility information.
 - Improved: [#16740] Allow staff patrol areas to be defined with individual tiles rather than groups of 4x4.
 - Improved: [#16764] [Plugin] Add hook 'map.save', called before the map is about is saved.
+- Improved: [#16925] The queue length of 1000 guests is lifted, and a warning for too long queues is added in its stead.
```
Comment (James103) [inline distribution/changelog.txt:38]:
```suggestion
- Improved: [#16925] The queue length of 1000 guests is lifted and a warning for too long queues is added instead.
```

**Comment by IntelOrca** [comment]
I have been looking at the park saving code. I think anyone who has saved a park within the duration this cheat was added will have that cheat serialised to their save file. Because it stores the number of searalised cheats, there is no compatibility issue. I think the only side effect will be that the next cheat that is added will instead get that value (but only for the parks saved during this short period).

**Comment by ZehMatt** [comment]
> I have been looking at the park saving code. I think anyone who has saved a park within the duration this cheat was added will have that cheat serialised to their save file. Because it stores the number of searalised cheats, there is no compatibility issue. I think the only side effect will be that the next cheat that is added will instead get that value (but only for the parks saved during this short period).

I recall saving the cheats as a pair, so it should be able to skip it and new cheats shouldn't influence old ones.

**Comment by Broxzier** [comment]
@IntelOrca Do you think we should implement a dummy cheat as a placeholder? That'll avoid issues with a future cheat. On the other hand, I don't expect there to be many park files with the cheat enabled (though that's speculation).

**Comment by IntelOrca** [comment]
@Broxzier Hmm, you might have to because, that cheat now needs to serialise / deserialise exactly the same number of bytes. Unfortunately this is using DataSerialiser (even in .park file), which saves everything as the size of the given type in big endian.

File: src/openrct2/entity/Peep.cpp [hunk 3]
```diff
@@ -1012,9 +1012,19 @@ void peep_problem_warnings_update()
              disgust_counter = 0, toilet_counter = 0, vandalism_counter = 0;
     uint8_t* warning_throttle = gPeepWarningThrottle;
 
+    int32_t inQueueCounter = 0;
+    int32_t tooLongQueueCounter = 0;
+    std::map<RideId, int32_t> queueComplainingGuestsMap;
```
Comment (CookiePLMonster) [inline src/openrct2/entity/Peep.cpp:1017]: Variables above are in `snake_case`, while this isn't. Might be worth changing to make it consistent?

**Comment by Broxzier** [comment]
@IntelOrca 4f34e849aed9a9d3588c1d52a2d21bc59275e7a4 reintroduces a cheat with the same ID to fix the deserialisation problem. If I understand the problem correctly, this is all that's needed. I have not tested this with an actual park file.

**Review by IntelOrca — APPROVED** [review]

**Pull request merged**